### DISCUSSION TITLE
Router taking query params

### DISCUSSION THREAD

**Discussion by youssefKadaouiAbbassi** [nylo-core/nylo#23] [Q&A] [answered]
Hey guys! @agordn52 @flutter-nylo @lpdevit @Abdulrasheed1729 

First of all, I wanted to thank you guys for the great job you are doing.

I wanted to ask if it is possible to create routes that take either query parameters or arguments.
Example: url/document/:id  or url/document?id=randomId
PS: I saw that Fluro (routing package) was capable of this. But I prefer your version of the routing system.
For a solution I think changing a bit the onGenerateRoute function to handle query params can do the job. Tell me if you want me to try it myself and submit a pull request.

Also, tell me if I can contribute in any way possible to the project.

Keep the great job 🔥!

**Comment by agordn52** [comment]
Hi @youssefKadaouiAbbassi, thanks for the kind message.

**Re. Query parameters**
That would be a cool addition to the framework. 

If you've got a working version with query parameters, send a PR and I'll check it out. 
I think it would improve the current routing system.

Thanks,
Anthony 👌

**Reply by youssefKadaouiAbbassi** [reply]
hi @agordn52 !

I just submitted a pull request on the nylo_core/support repo [here](https://github.com/nylo-core/support/pull/20) that adds the query parameters support. 
I hope the tests go well!

### To test:
go to a route with something like: url/route?arg=1
on your page or widget state call widget.queryParams() as you would call widget.data() for passed arguments.

### The next step: 
 routes with parameters:  (url/route/:id)


Thanks again,
Youssef.

**Reply by agordn52** [reply]
@youssefKadaouiAbbassi 🧙‍♀️ 

Great idea, I'll check out the code later :) 
Thanks for contributing man!

Anthony

**Comment by agordn52** [comment, marked as answer]
PR [20](https://github.com/nylo-core/support/pull/20) merged, thanks @youssefKadaouiAbbassi  🚀
If you have any other ideas, feel free to drop them here. We can collaborate.
